<?xml version="1.0"?>
<launch>

  <arg name="robot_namespace"   default="qev2"/>
  <arg name="updateRate" 				default="10"/>
  <arg name="M_PI"              value="3.1415926535897931" />

  <node name="lidar_clustering" pkg="qutms_autonomous_nodes" type="lidar_clustering_node" respawn="true" output="screen" >
    
    <param name="inputTopic" type="string" value="/$(arg robot_namespace)/horizon_lidar/pointcloud" />
    <param name="outputTopic" type="string" value="/$(arg robot_namespace)/lidar_clustering_node/pointcloud_filtered" />

    <!-- Parameters for filtering ground -->
    <param name="filterZMax" type="double" value="5.0" />
    <param name="filterZMin" type="double" value="-0.08" />

    <!-- Parameter for voxel grid filter -->
    <param name="leafSize" type="double" value="0.01" />

    <!-- Parameters for clustering -->
    <param name="rangeMax" type="double" value="25" />
    <param name="openingDegreeMax" type="double" value="$(arg M_PI)" />
    <param name="openingDegreeMin" type="double" value="0" />
    <param name="thresholdRadius" type="double" value="0.1" />
    <param name="noiseThreshold" type="int" value="3" />

    <!-- Parameters for LShapeFitting -->
    <param name="lShapeFittingEnabled" type="bool" value="true" />
    <param name="amountOfSteps" type="int" value="90" />
    <param name="degreeThreshold" type="double" value="0.001" />

  </node>

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find qutms_autonomous_nodes)/rviz/lidar_clustering.rviz" />

</launch>